app/lib/llm_clients/openai/client.rb

```diff
@@ -1,77 +1,74 @@
-# rubocop:disable all
+require "openai"
+
 module LlmClients
   module Openai
-    class Client < Client
-      def call(prompt)
-        @stats = new_stats
-        request = request(prompt)
+    class Client < LlmClients::Client
+      include ChatRequestHelper
+      include EmbeddingRequestHelper
+      include CompletionRequestHelper
 
-        Net::HTTP.start(@uri.hostname, @uri.port, use_ssl: @uri.scheme == "https") do |http|
-          http.request(request) do |response|
-            raise response_error_for(response) unless response.code.to_i >= 200 && response.code.to_i < 300
+      NETWORK_TIMEOUT = 8
 
-            stats[:start_time] = current_time
-            stats[:first_token_time] = current_time
-            body = response.read_body
-            content = ""
-            each_message(body) do |message|
-              @stats[:tokens] += 1
-              content += message if message
-            end
+      def complete(prompt, &)
+        complete_request(prompt, &)
+      end
 
-            yield content
+      def chat(message, &)
+        chat_request(ChatMessageHelper.new(message).chat_history, &)
+      end
 
-            calculate_stats
-          end
-        end
+      def embed(content)
+        embedding_request(content)
+      end
+
+      # Callback for instrumenting request via Faraday middleware used by OpenAI API gem
+      def instrument(_name, env)
+        tmp = yield
```

Comment:
Would calling this `response` be more clear?

Reply:
Yes it would. :-)